src/legacy/core_plugins/interpreter/public/functions/tagcloud.js

```diff
@@ -28,16 +28,94 @@ export const tagcloud = () => ({
     ],
   },
   help: i18n.translate('interpreter.functions.tagcloud.help', {
-    defaultMessage: 'Tagcloud visualization'
+    defaultMessage: 'Tagcloud visualization element'
   }),
   args: {
-    visConfig: {
-      types: ['string', 'null'],
-      default: '"{}"',
+    scale: {
+      types: ['string'],
+      default: 'linear',
+      options: ['linear', 'log', 'square root'],
+      help: i18n.translate('interpreter.functions.tagcloud.scale.help', {
+        defaultMessage: 'Scale to determine font size of a word'
+      }),
+    },
+    orientation: {
+      types: ['string'],
+      default: 'single',
+      options: ['single', 'right angled', 'multiple'],
+      help: i18n.translate('interpreter.functions.tagcloud.orientation.help', {
+        defaultMessage: 'Orientation of words inside tagcloud'
+      }),
+    },
+    minFontSize: {
+      types: ['number'],
+      default: 18,
+    },
+    maxFontSize: {
+      types: ['number'],
+      default: 72
+    },
+    showLabel: {
+      types: ['boolean'],
+      default: true,
+    },
+    metric: {
+      types: ['string', 'number'],
+      help: i18n.translate('interpreter.functions.tagcloud.metric.help', {
+        defaultMessage: 'Column in your dataset to use as a metric (either column index or column name)'
+      }),
+    },
+    bucket: {
+      types: ['string', 'number'],
+      help: i18n.translate('interpreter.functions.tagcloud.bucket.help', {
+        defaultMessage: 'Column in your dataset to use as a bucket (either column index or column name)'
+      }),
     },
+    bucketFormat: {
+      types: ['string'],
+      default: 'string'
+    },
+    bucketFormatParams: {
+      types: ['string'],
+      default: '"{}"',
+    }
   },
   fn(context, args) {
-    const visConfig = JSON.parse(args.visConfig);
+    const metricAccessor = Number.isInteger(args.metric) ?
+      args.metric :
+      context.columns.find(c => c.id === args.metric);
+    if (metricAccessor === undefined) {
+      throw new Error('invalid column name');
```

Comment:
The user sees these errors, giving them a little more information would likely help them be successful. Right now they get the same message with an invalid metric or bucket, at the very least you should tell them which argument is causing the error. Telling them why it's failing (unable to find column foo, or no column at index X) would be ideal.

Reply:
also these all need to be translated